<a href="https://colab.research.google.com/github/HakumenWorld/data-science-2026/blob/main/Pertemuan2_Febi_Kristiana_240401010231.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama: Febi Kristiana
NIM: 240401010231
Prodi: Informatika

In [10]:
#Impor library: import pandas as pd dan import numpy as np
import pandas as pd
import numpy as np

In [16]:
import pandas as pd
#Memuat dataset Titanic menggunakan pd.read_csv()
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


Q1 Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?

In [12]:
# Menampilkan jumlah baris dan kolom
shape = df.shape
print(f"Jumlah baris dan kolom: {shape}\n")

# Menampilkan daftar kolom yang tersedia
columns = df.columns
print(f"Kolom yang tersedia: \n{columns.tolist()}")

Jumlah baris dan kolom: (891, 12)

Kolom yang tersedia: 
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Pada dataset Titanic, terdapat 891 baris dan 12 kolom. Kolom-kolom yang tersedia antara lain:
- 'PassengerId', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked', dan 'Survived'.
Kolom-kolom ini memberikan informasi tentang penumpang Titanic, termasuk status keselamatan ('Survived'), kelas tiket ('Pclass'), informasi pribadi seperti usia dan jenis kelamin ('Age' dan 'Sex'), serta data lain yang relevan.

Q2 Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%)
per kolom?

In [17]:
# Menampilkan kolom yang memiliki missing values
missing_values = df.isnull().sum()
missing_percent = (df.isnull().sum() / df.shape[0]) * 100

# Menampilkan kolom dengan missing values dan proporsinya
print("Jumlah missing values per kolom:\n", missing_values)
print("\nProporsi missing values per kolom (%):\n", missing_percent)

pd.DataFrame({
    'Jumlah Missing': missing_values,
    'Persentase (%)': missing_percent
})

Jumlah missing values per kolom:
 PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Proporsi missing values per kolom (%):
 PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64


,Jumlah Missing,Persentase (%)
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


Beberapa kolom dalam dataset Titanic memiliki missing values. Kolom yang paling banyak missing values-nya adalah 'Cabin', yang hampir 77% datanya hilang. Kolom 'Age' juga memiliki sejumlah missing values, yaitu sekitar 20%. Kolom lain seperti 'Embarked' memiliki persentase missing yang lebih kecil.
Persentase missing values ini menunjukkan seberapa besar bagian data yang hilang di setiap kolom, dan ini penting untuk menentukan bagaimana kita akan menangani missing values dalam analisis selanjutnya.

Q3 Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy
atau Pandas.

In [21]:
# Menghitung persentase penumpang yang selamat
survived_percentage = (df['Survived'].mean()) * 100
print(f"Persentase penumpang yang selamat: {survived_percentage:.2f}%")

Persentase penumpang yang selamat: 38.38%


Persentase penumpang yang selamat pada Titanic adalah sekitar 38%. Ini dihitung dengan mengambil rata-rata nilai dari kolom 'Survived', di mana 1 menunjukkan bahwa penumpang selamat dan 0 berarti penumpang tidak selamat. Hasil ini memberikan gambaran tentang tingkat kelangsungan hidup penumpang selama kejadian tersebut.

Q4 Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata- rata usianya?

In [25]:
#Filter penumpang wanita dari kelas 1
female_class_1 = df[(df['Pclass'] == 1) & (df['Sex'] == 'female')]

# Menghitung jumlah dan rata-rata usia
num_female_class_1 = female_class_1.shape[0]
average_age_female_class_1 = female_class_1['Age'].mean()

print(f"Jumlah penumpang wanita di kelas 1: {num_female_class_1}")
print(f"Rata-rata usia penumpang wanita di kelas 1: {average_age_female_class_1:.2f}")

Jumlah penumpang wanita di kelas 1: 94
Rata-rata usia penumpang wanita di kelas 1: 34.61



Terdapat 94 penumpang wanita dari kelas 1 pada dataset Titanic. Rata-rata usia mereka adalah sekitar 34,6 tahun. Data ini memberikan gambaran tentang demografi penumpang wanita di kelas 1, yang cenderung lebih dewasa dibandingkan dengan kelompok penumpang wanita di kelas lain.

Q5 Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas
penumpang. Kelas mana yang paling tinggi?

In [34]:
# Menghitung tingkat keselamatan per kelas penumpang menggunakan groupby
safety_by_class = df.groupby('Pclass').agg(
    Total = ('PassengerId', 'count'),
    Selamat = ('Survived', 'mean')
)
print("Tingkat keselamatan per kelas:\n", safety_by_class)

# Menentukan kelas dengan tingkat keselamatan tertinggi
highest_safety_class = safety_by_class.idxmax()
print(f"Kelas dengan tingkat keselamatan tertinggi: {highest_safety_class}")

Tingkat keselamatan per kelas:
         Total   Selamat
Pclass                 
1         216  0.629630
2         184  0.472826
3         491  0.242363
Kelas dengan tingkat keselamatan tertinggi: Total      3
Selamat    1
dtype: int64


Tingkat keselamatan per kelas penumpang dihitung dengan mengelompokkan data berdasarkan 'Pclass' dan menghitung rata-rata nilai 'Survived' untuk setiap kelompok. Dari hasil ini, kita menemukan bahwa kelas 1 memiliki tingkat keselamatan tertinggi, yaitu sekitar 62%. Hal ini menunjukkan bahwa penumpang kelas 1 memiliki peluang lebih besar untuk selamat dibandingkan dengan penumpang kelas 2 dan 3.

Q6 Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik.
Sertakan interpretasi singkat.

In [35]:
# Analisis tambahan: Menghitung rata-rata tarif berdasarkan jenis kelamin
average_fare_by_sex = df.groupby('Sex')['Fare'].mean()
print("Rata-rata tarif berdasarkan jenis kelamin:\n", average_fare_by_sex)

Rata-rata tarif berdasarkan jenis kelamin:
 Sex
female    44.479818
male      25.523893
Name: Fare, dtype: float64


Dari analisis rata-rata tarif berdasarkan jenis kelamin, kita melihat bahwa penumpang wanita membayar rata-rata tarif yang lebih tinggi daripada penumpang pria. Rata-rata tarif penumpang wanita adalah sekitar 44,5, sementara rata-rata tarif penumpang wanita adalah sekitar 25. Hal ini bisa jadi mencerminkan perbedaan kelas atau jenis tiket yang dibeli oleh kedua kelompok, serta perbedaan dalam aksesibilitas layanan.